File: objc/Project2/Project2/Converter.m
//
//  Converter.m
//  Project2
//
//  Created by Jinwoo Kim on 2/24/21.
//

#import "Converter.h"

@implementation Converter
- (double)convertToCelsiusFromFahrenheit:(double)fahrenheit {
//    return (fahrenheit - 32) * 5 / 9;
    NSMeasurement *fahrenheitMeasurement = [[NSMeasurement alloc] initWithDoubleValue:fahrenheit unit:[NSUnitTemperature fahrenheit]];
    NSMeasurement *celsiusMeasurement = [fahrenheitMeasurement measurementByConvertingToUnit:[NSUnitTemperature celsius]];
    return celsiusMeasurement.doubleValue;
}
@end

File: swift/Project2/Project2/Converter.swift
//
//  Converter.swift
//  Project2
//
//  Created by Jinwoo Kim on 2/22/21.
//

import Foundation

struct Converter {
    func convertToCelsius(fahrenheit: Double) -> Double {
//        return (fahrenheit - 32) * 5 / 9
        let fahrenheit = Measurement(value: fahrenheit, unit: UnitTemperature.fahrenheit)
        let celsius = fahrenheit.converted(to: UnitTemperature.celsius)
        return celsius.value
    }
}
